api/app/weather_models/precip_rdps_model.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import numpy
+from numba import vectorize
+from app.utils.s3 import read_into_memory
+from app.weather_models import ModelEnum
+from app.weather_models.rdps_filename_marshaller import compose_computed_precip_rdps_key
+
+
+@dataclass
+class TemporalPrecip:
+    """Precip values at certain times"""
+
+    timestamp: str
+    precip_amount: float | numpy.ndarray
+
+    def is_after(self, other) -> bool:
+        return self.timestamp > other.timestamp
+
+
+RDPS_S3_PREFIX = f"weather_models/{ModelEnum.RDPS.lower()}/"
+
+
+async def generate_24_hour_accumulating_precip_raster(current_time: datetime):
+    """
+    Given a UTC datetime, grab the raster for that date
+    and the date for 24 hours before to compute the difference.
+    """
+    (yesterday_key, today_key) = get_raster_keys_to_diff(current_time)
+    day_data = await read_into_memory(f"{RDPS_S3_PREFIX}/{today_key}")
```

Comment:
I think we need some logic here to handle the case where the `earlier_key`/`yesterday_key` is None.

Reply:
Hopefully we can just read the raster at `today_key` and return the data from that?

Reply:
Done in: https://github.com/bcgov/wps/pull/3749/commits/2fd40f78d3c5de51a3ed4d624fd21d70d78ee3c5